### ISSUE TITLE
Get java.lang.NoSuchMethodError: org.joda.time.DateTime.now()Lorg/joda/time/DateTime when trying to generate code with gradle-swagger-generator-plugin

### ISSUE THREAD

**Issue by Plinzen** [int128/gradle-swagger-generator-plugin#65]
Hi, 

I have a swagger `OpenAPI YAML` file and I want to generate `Java` using the plugin.

This is my `config.json`

    {
      "useRxJava2": true,
      "java8": false,
      "library": "retrofit2",
      "dateLibrary": "joda",
      "hideGenerationTimestamp": true,
      "modelPackage": "backend.model",
      "apiPackage": "backend.api",
      "invokerPackage": "backend"
    }

and this is my `build.gradle`

    plugins {
        id 'java'
        id 'org.hidetake.swagger.generator' version '2.8.0'
    }
    
    repositories {
        jcenter()
    }
    
    dependencies {
        // Add dependency for Swagger Codegen CLI
        swaggerCodegen 'io.swagger:swagger-codegen-cli:2.2.3'
        compile 'io.swagger:swagger-annotations:1.5.10'
        compile "joda-time:joda-time:$jodaTimeVersion"
    }
    
    swaggerSources {
        backend {
            inputFile = file('api.yaml')
            code {
                language = 'java'
                configFile = file('config.json')
                components = ['models', 'apis']
            }
        }
    }
    
    compileJava.dependsOn swaggerSources.keycore.code
    sourceSets.main.java.srcDir swaggerSources.keycore.code.outputDir

When I try to generate the code via gradle:

    ./gradlew clean generateSwaggerCodeBackend

I receive the following exception during build:

    :swagger-java-client:resolveSwaggerTemplate UP-TO-DATE
    :swagger-java-client:generateSwaggerCodeBackend FAILED
    
    FAILURE: Build failed with an exception.
    
    * What went wrong:
    Execution failed for task ':backend-swagger-java-client:generateSwaggerCodeBackend'.
    > org.joda.time.DateTime.now()Lorg/joda/time/DateTime;
    
    * Try:
    Run with --info or --debug option to get more log output.
    
    * Exception is:
    org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':keycore-swagger-java-client:generateSwaggerCodeKeycore'.
            at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:84)
            at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:55)
            at org.gradle.api.internal.tasks.execution.SkipUpToDateTaskExecuter.execute(SkipUpToDateTaskExecuter.java:62)
            at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:58)
            at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFilesTaskExecuter.java:88)
            at org.gradle.api.internal.tasks.execution.ResolveTaskArtifactStateTaskExecuter.execute(ResolveTaskArtifactStateTaskExecuter.java:46)
            at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:51)
            at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:54)
            at org.gradle.api.internal.tasks.execution.ExecuteAtMostOnceTaskExecuter.execute(ExecuteAtMostOnceTaskExecuter.java:43)
            at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:34)
            at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker$1.execute(DefaultTaskGraphExecuter.java:236)
            at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker$1.execute(DefaultTaskGraphExecuter.java:228)
            at org.gradle.internal.Transformers$4.transform(Transformers.java:169)
            at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:106)
            at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:61)
            at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:228)
            at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:215)
            at org.gradle.execution.taskgraph.AbstractTaskPlanExecutor$TaskExecutorWorker.processTask(AbstractTaskPlanExecutor.java:77)
            at org.gradle.execution.taskgraph.AbstractTaskPlanExecutor$TaskExecutorWorker.run(AbstractTaskPlanExecutor.java:58)
            at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor.process(DefaultTaskPlanExecutor.java:32)
            at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter.execute(DefaultTaskGraphExecuter.java:113)
            at org.gradle.execution.SelectedTaskExecutionAction.execute(SelectedTaskExecutionAction.java:37)
            at org.gradle.execution.DefaultBuildExecuter.execute(DefaultBuildExecuter.java:37)
            at org.gradle.execution.DefaultBuildExecuter.access$000(DefaultBuildExecuter.java:23)
            at org.gradle.execution.DefaultBuildExecuter$1.proceed(DefaultBuildExecuter.java:43)
            at org.gradle.execution.DryRunBuildExecutionAction.execute(DryRunBuildExecutionAction.java:32)
            at org.gradle.execution.DefaultBuildExecuter.execute(DefaultBuildExecuter.java:37)
            at org.gradle.execution.DefaultBuildExecuter.execute(DefaultBuildExecuter.java:30)
            at org.gradle.initialization.DefaultGradleLauncher$3.execute(DefaultGradleLauncher.java:196)
            at org.gradle.initialization.DefaultGradleLauncher$3.execute(DefaultGradleLauncher.java:193)
            at org.gradle.internal.Transformers$4.transform(Transformers.java:169)
            at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:106)
            at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:56)
            at org.gradle.initialization.DefaultGradleLauncher.doBuildStages(DefaultGradleLauncher.java:193)
            at org.gradle.initialization.DefaultGradleLauncher.doBuild(DefaultGradleLauncher.java:119)
            at org.gradle.initialization.DefaultGradleLauncher.run(DefaultGradleLauncher.java:102)
            at org.gradle.launcher.exec.GradleBuildController.run(GradleBuildController.java:71)
            at org.gradle.tooling.internal.provider.ExecuteBuildActionRunner.run(ExecuteBuildActionRunner.java:28)
            at org.gradle.launcher.exec.ChainingBuildActionRunner.run(ChainingBuildActionRunner.java:35)
            at org.gradle.launcher.exec.InProcessBuildActionExecuter.execute(InProcessBuildActionExecuter.java:41)
            at org.gradle.launcher.exec.InProcessBuildActionExecuter.execute(InProcessBuildActionExecuter.java:26)
            at org.gradle.tooling.internal.provider.ContinuousBuildActionExecuter.execute(ContinuousBuildActionExecuter.java:75)
            at org.gradle.tooling.internal.provider.ContinuousBuildActionExecuter.execute(ContinuousBuildActionExecuter.java:49)
            at org.gradle.tooling.internal.provider.ServicesSetupBuildActionExecuter.execute(ServicesSetupBuildActionExecuter.java:44)
            at org.gradle.tooling.internal.provider.ServicesSetupBuildActionExecuter.execute(ServicesSetupBuildActionExecuter.java:29)
            at org.gradle.launcher.daemon.server.exec.ExecuteBuild.doBuild(ExecuteBuild.java:67)
            at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:36)
            at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
            at org.gradle.launcher.daemon.server.exec.WatchForDisconnection.execute(WatchForDisconnection.java:47)
            at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
            at org.gradle.launcher.daemon.server.exec.ResetDeprecationLogger.execute(ResetDeprecationLogger.java:26)
            at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
            at org.gradle.launcher.daemon.server.exec.RequestStopIfSingleUsedDaemon.execute(RequestStopIfSingleUsedDaemon.java:34)
            at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
            at org.gradle.launcher.daemon.server.exec.ForwardClientInput$2.call(ForwardClientInput.java:74)
            at org.gradle.launcher.daemon.server.exec.ForwardClientInput$2.call(ForwardClientInput.java:72)
            at org.gradle.util.Swapper.swap(Swapper.java:38)
            at org.gradle.launcher.daemon.server.exec.ForwardClientInput.execute(ForwardClientInput.java:72)
            at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
            at org.gradle.launcher.daemon.server.exec.LogAndCheckHealth.execute(LogAndCheckHealth.java:55)
            at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
            at org.gradle.launcher.daemon.server.exec.LogToClient.doBuild(LogToClient.java:60)
            at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:36)
            at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
            at org.gradle.launcher.daemon.server.exec.EstablishBuildEnvironment.doBuild(EstablishBuildEnvironment.java:72)
            at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:36)
            at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
            at org.gradle.launcher.daemon.server.exec.StartBuildOrRespondWithBusy$1.run(StartBuildOrRespondWithBusy.java:50)
            at org.gradle.launcher.daemon.server.DaemonStateCoordinator$1.run(DaemonStateCoordinator.java:297)
            at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:54)
            at org.gradle.internal.concurrent.StoppableExecutorImpl$1.run(StoppableExecutorImpl.java:40)
    Caused by: java.lang.NoSuchMethodError: org.joda.time.DateTime.now()Lorg/joda/time/DateTime;
            at io.swagger.codegen.DefaultGenerator.configureGeneratorProperties(DefaultGenerator.java:136)
            at io.swagger.codegen.DefaultGenerator.generate(DefaultGenerator.java:687)
            at io.swagger.codegen.cmd.Generate.run(Generate.java:285)
            at io.swagger.codegen.SwaggerCodegen.main(SwaggerCodegen.java:35)
            at org.hidetake.gradle.swagger.generator.SwaggerCodegenExecutor.execute(SwaggerCodegenExecutor.groovy:76)
            at org.hidetake.gradle.swagger.generator.SwaggerCodegenExecutor$execute$0.call(Unknown Source)
            at org.hidetake.gradle.swagger.generator.GenerateSwaggerCode.exec(GenerateSwaggerCode.groovy:53)
            at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:73)
            at org.gradle.api.internal.project.taskfactory.DefaultTaskClassInfoStore$StandardTaskAction.doExecute(DefaultTaskClassInfoStore.java:141)
            at org.gradle.api.internal.project.taskfactory.DefaultTaskClassInfoStore$StandardTaskAction.execute(DefaultTaskClassInfoStore.java:134)
            at org.gradle.api.internal.project.taskfactory.DefaultTaskClassInfoStore$StandardTaskAction.execute(DefaultTaskClassInfoStore.java:123)
            at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:632)
            at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:615)
            at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:95)
            at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:76)
            ... 70 more
    
    
    BUILD FAILED

Basically it is pointing me to an 

    Caused by: java.lang.NoSuchMethodError: org.joda.time.DateTime.now()Lorg/joda/time/DateTime;

But I have no idea, how to fix this. I tried several options of adding joda-time to the dependencies section, but non of them worked:

    compile "joda-time:joda-time:$jodaTimeVersion"
    swaggerCodegen "joda-time:joda-time:$jodaTimeVersion"

I also asked this question on SO for your reference:
https://stackoverflow.com/questions/46907653/get-java-lang-nosuchmethoderror-org-joda-time-datetime-nowlorg-joda-time-date

Thank you for your help!

**Comment by int128** [comment]
Unfortunately I could not reproduce your problem. Would you run `./gradlew buildEnvironment`?

The dependency `joda-time` is included in my environment as follows:

```
:template-project:buildEnvironment

------------------------------------------------------------
Project :template-project
------------------------------------------------------------

classpath
\--- gradle.plugin.org.hidetake:gradle-swagger-generator-plugin:2.8.0
     +--- com.github.fge:json-schema-validator:2.2.6
     |    +--- com.google.code.findbugs:jsr305:3.0.0
     |    +--- joda-time:joda-time:2.3
     |    +--- com.googlecode.libphonenumber:libphonenumber:6.2
     |    +--- com.github.fge:json-schema-core:1.2.5
     |    |    +--- com.github.fge:uri-template:0.9
     |    |    |    +--- com.github.fge:msg-simple:1.1
     |    |    |    |    +--- com.github.fge:btf:1.2
     |    |    |    |    |    \--- com.google.code.findbugs:jsr305:2.0.1 -> 3.0.0
     |    |    |    |    \--- com.google.code.findbugs:jsr305:2.0.1 -> 3.0.0
     |    |    |    +--- com.google.guava:guava:16.0.1
     |    |    |    \--- com.google.code.findbugs:jsr305:2.0.1 -> 3.0.0
     |    |    +--- com.github.fge:jackson-coreutils:1.8
     |    |    |    +--- com.github.fge:msg-simple:1.1 (*)
     |    |    |    +--- com.fasterxml.jackson.core:jackson-databind:2.2.3
     |    |    |    |    +--- com.fasterxml.jackson.core:jackson-annotations:2.2.3
     |    |    |    |    \--- com.fasterxml.jackson.core:jackson-core:2.2.3
     |    |    |    +--- com.google.guava:guava:16.0.1
     |    |    |    \--- com.google.code.findbugs:jsr305:2.0.1 -> 3.0.0
     |    |    +--- org.mozilla:rhino:1.7R4
     |    |    \--- com.google.code.findbugs:jsr305:2.0.1 -> 3.0.0
     |    +--- javax.mail:mailapi:1.4.3
     |    |    \--- javax.activation:activation:1.1
     |    \--- net.sf.jopt-simple:jopt-simple:4.6
     \--- com.fasterxml.jackson.dataformat:jackson-dataformat-yaml:2.2.3
          \--- com.fasterxml.jackson.core:jackson-core:2.2.3
```

**Comment by Plinzen** [comment]
I works again on my side after upgrading my IDE to Android Studio 3. Thanks for your help.
